### ISSUE TITLE
Notifications are not arriving on iOS (App Store version) since update 2.3.0

### ISSUE THREAD

**Issue by judari** [phonegap/phonegap-plugin-push#2857]
:ambulance: :apple: :arrow_up:

Hi everyone,

We are working on an Ionic project and we are using this plugin. The plugin and the notifications had been working perfectly until we updated to 2.3.0 version.

In one point, we had to update to 2.3.0 version because of iOS 13. In that iOS update, the token changed to something like this: {length=32,bytes=....}. Whit the 2.3.0 version that problem was fixed. However, another problem appeared.

Now, the plugin and the notifications are working perfectly in Android (development and production version) and iOS (BUT ONLY IN DEVELOPMENT VERSION).

Currently, the notifications are not arriving in the iOS App Store version. We already tried everything. Also, we checked all the certificates, identifiers and profiles, but all seems to be perfect (photos attached).

I hope all of you could help us. All options are welcome.

Regards.

NOTE: There is no problem with the server. Notifications are being sent correctly.

### Expected Behaviour

Notifications arriving on iOS App Store version

### Actual Behaviour

Notifications are not arriving on iOS (App Store version)

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)

iOS 13.2.2

#### Cordova CLI version and cordova platform version

    cordova --version            8.0.0
    cordova platform version ios              5.0.1

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push            2.3.0

#### Certificates, identifiers and profiles screenshots

![1-Profiles](https://user-images.githubusercontent.com/7573689/68877618-5bba7080-06d4-11ea-8da5-c1828fd5d9a0.png)
![2-Identifiers](https://user-images.githubusercontent.com/7573689/68877620-5c530700-06d4-11ea-9e68-be4de55d117a.png)
![3-APNSDevelopment](https://user-images.githubusercontent.com/7573689/68877621-5c530700-06d4-11ea-8772-fc2c369ec16f.png)
![4-APNSProduction](https://user-images.githubusercontent.com/7573689/68877623-5c530700-06d4-11ea-8c40-95167ff06c49.png)
![5-DistributionCert](https://user-images.githubusercontent.com/7573689/68877625-5c530700-06d4-11ea-9490-bf612b4a3252.png)
![6-DevelopmentCert](https://user-images.githubusercontent.com/7573689/68877626-5c530700-06d4-11ea-91fd-f4118aa5ff75.png)
![7-ProjectConfig](https://user-images.githubusercontent.com/7573689/68877627-5ceb9d80-06d4-11ea-907f-cd32906a628c.png)

**Comment by rychter** [comment]
I have a setting in azure notification hub that allows me to switch between sandbox and prod version. Maybe in your notification hub it is switched to sandbox (unsigned)?

**Comment by judari** [comment]
Hi rychter. We already checked the server side and all is correctly. The weird aspect is that the notifications were working perfectly with the older plugin version.

**Comment by zdm** [comment]
Not work for me too.
Don't know what is the problem, I didn't changed anything, just rebuild app.
Previously push notifications worked on ios and android, now only for android.

**Comment by sammyrsvp** [comment]
Not working for me either. Same as @zdm.  Any one tried cordova version 9?

**Comment by sammyrsvp** [comment]
Tested 2.2.3 on iOS 13, I can see a popup window: "<App> would like to send you notifications  with 2 buttons  "Do not Allow" and "Allow". 
But on 2.3.0, there is no popup.

**Comment by zdm** [comment]
I recreated certificate, provisioning profile and APNS key - and is start working.

**Comment by sammyrsvp** [comment]
Hi @zdm thank you very much for your update. Did you receive a popup window when the app is installed an opened for the first time: "Would you like the app to send you notifications?  2 buttons <Do not Allow> and <Allow>"

**Comment by webteambel** [comment]
@judari  any update of this ? we have the same problem and we didn't find a solution yet, we generate new certificates and provisioning profile ... and we use gateway.push.apple.com:2195 but it's not working :(

**Comment by judari** [comment]
Hi everyone,

Yes, we managed to solve it. The problem was that we were creating the certificate using the add certificate option (see picture 1) and then we linked it to the identifier. This is not the way to do it, this can generate the issue that is in this blog. The correct way to generate the certificate is doing it from the identifier using the "Create Certificate" option (see picture 2 and 3). I hope this solution helps you.

Regards

Picture 1
![Screen Shot 2020-02-24 at 11 05 41 AM](https://user-images.githubusercontent.com/7573689/75169719-0809eb00-56f7-11ea-80d2-1d158142cbab.png)

Picture 2
![Screen Shot 2020-02-24 at 11 05 59 AM](https://user-images.githubusercontent.com/7573689/75169965-6636ce00-56f7-11ea-9c94-8945db833c89.png)

Picture 3
![Screen Shot 2020-02-24 at 11 06 13 AM](https://user-images.githubusercontent.com/7573689/75169781-22dc5f80-56f7-11ea-8e55-abbc04168d7c.png)
